addons/mail/static/tests/mail_utils.test.js

```diff
@@ -184,3 +185,17 @@ test("url with number in subdomain", async () => {
         ".o-mail-Message a:contains(https://www.45017478-master-all.runbot134.odoo.com/web)"
     );
 });
+
+test("isSequential doesn't execute intermediate call.", async () => {
+    const sequential = useSequential();
+    let index = 0;
+    const sequence = () => {
+        index++;
+        return sequential(async () => {
+            expect.step(index.toString());
+            return new Promise((r) => setTimeout(r, 1));
+        });
+    };
+    await Promise.all([sequence(), sequence(), sequence(), sequence(), sequence()]);
+    expect(["1", "5"]).toVerifySteps();
```

Comment:
It's not obvious from this test which one is executed:

- the counter is incremented synchronously immediately, so any step past the first would probably return 5
- it's not clear what becomes of the intermediate promise from the test (pending? probably not as the await is unlocked, they resolve what?)